ge/org.osate.ge/src/org/osate/ge/internal/ui/tools/ToolUtil.java

```diff
@@ -80,4 +108,234 @@ public static Context findContextExcludeOwner(final BusinessObjectContext boc,
+		}
+
+		final XtextResource resource = getXtextResource(resourceSet, eObjectToValidate.eResource().getURI());
+		loadResource(resource, serializedSrc.get());
+
+		final Builder<DiagnosticWrapper> diagnostics = Stream.builder();
+
+		// Concrete Syntax Validation
+		final Stream<DiagnosticWrapper> syntaxDiagnostics = resource.validateConcreteSyntax().stream()
+				.filter(diagnostic -> isErrorOrWarning(diagnostic.getSeverity()))
+				.map(diagnostic -> diagnosticBuilder.build(diagnostic));
+		addToBuilder(diagnostics, syntaxDiagnostics);
+
+		final EObject serializedObject = resourceSet.getEObject(EcoreUtil.getURI(eObjectToValidate), true);
+		final Diagnostic validationDiagnostic = Diagnostician.INSTANCE.validate(serializedObject,
+				Collections.singletonMap(Diagnostician.VALIDATE_RECURSIVELY, true));
+		if (isErrorOrWarning(validationDiagnostic.getSeverity())) {
+			// Validation Diagnostics
+			final Stream<DiagnosticWrapper> qualifiedDiagnostics = getDiagnosticDescendants(validationDiagnostic)
+					.filter(diagnostic -> diagnostic instanceof FeatureBasedDiagnostic)
+					.map(diagnostic -> diagnosticBuilder.build(diagnostic));
+			addToBuilder(diagnostics, qualifiedDiagnostics);
+		}
+
+		// Errors
+		final Stream<DiagnosticWrapper> resourceErrors = getResourceDiagnostics(resource.getErrors(),
+				diagnosticBuilder,
+				Diagnostic.ERROR);
+		addToBuilder(diagnostics, resourceErrors);
+
+		// Warnings
+		final Stream<DiagnosticWrapper> resourceWarnings = getResourceDiagnostics(resource.getWarnings(),
+				diagnosticBuilder,
+				Diagnostic.WARNING);
+		addToBuilder(diagnostics, resourceWarnings);
+
+		return diagnostics.build().collect(Collectors.toSet());
+	}
+
+	private static void addToBuilder(final Builder<DiagnosticWrapper> builder,
+			final Stream<DiagnosticWrapper> diagnosticsToAdd) {
+		diagnosticsToAdd.forEach(diagnostic -> builder.add(diagnostic));
+	}
+
+	private static class DiagnosticBuilder {
+		private final String msgPrefix;
+
+		public DiagnosticBuilder(final String msgPrefix) {
+			this.msgPrefix = msgPrefix;
+		}
+
+		public DiagnosticWrapper build(final int severity, final String msg) {
+			return new DiagnosticWrapper(severity, getDetailedDiagnosticMessage(this.msgPrefix, msg));
+		}
+
+		public DiagnosticWrapper build(final Diagnostic diagnostic) {
+			return new DiagnosticWrapper(diagnostic.getSeverity(),
+					getDetailedDiagnosticMessage(this.msgPrefix, diagnostic.getMessage()));
+		}
+
+		public DiagnosticWrapper build(final int severity,
+				final org.eclipse.emf.ecore.resource.Resource.Diagnostic diagnostic) {
+			return new DiagnosticWrapper(severity,
+					getDetailedDiagnosticMessage(this.msgPrefix, diagnostic.getMessage()));
+		}
+
+		private static String getDetailedDiagnosticMessage(final String prefix, final String message) {
+			return new StringBuilder(prefix).append(message).toString();
+		}
+	}
+
+	private static class DiagnosticWrapper extends BasicDiagnostic {
```

Comment:
Add a javadoc comment describing the purpose of this class. Rename if there is a better name. It is not clear to me why this exists.

Reply:
This exists so duplicate error messages do not return.  For instance, when name conflicts occur, two errors are returned.  One error for the flow being created and one for the flow that exists.  This is this way the validation works, but it did seem a little redundant.